services/motion/motion.go

```diff
@@ -43,6 +43,19 @@ type PlanHistoryReq struct {
 	Extra       map[string]interface{}
 }
 
+// MoveReq describes the request to the Move interface method.
+type MoveReq struct {
+	// ComponentName of the component to move
+	ComponentName resource.Name
+	// Goal destination the component should be moved to
+	Destination *referenceframe.PoseInFrame
```

Comment:
Not sure how I feel about this being a pointer.  We typically pass them around as pointers but that means the default value for this struct becomes nil, which is unacceptable.  For all the other arguments its fine if they are nil because they are optional.

Reply:
Per the conversation with @biotinker earlier in this draft, I'm going to leave it as a pointer and error if nil is passed.  

@nfranczak